kura/org.eclipse.kura.core.status/src/main/java/org/eclipse/kura/core/status/CloudConnectionStatusServiceImpl.java

```diff
@@ -37,227 +37,226 @@
 
 public class CloudConnectionStatusServiceImpl implements CloudConnectionStatusService {
 
-	private static final String STATUS_NOTIFICATION_URL	= "ccs.status.notification.url";
-	
+	private static final String STATUS_NOTIFICATION_URL = "ccs.status.notification.url";
+
 	private static final Logger s_logger = LoggerFactory.getLogger(CloudConnectionStatusServiceImpl.class);
-	
+
 	private SystemService m_systemService;
-	private GPIOService m_GPIOService;
-	
-	private int LedIndex = -1;
-	private KuraGPIOPin notificationLED;
-	
-	private ExecutorService notificationExecutor;
-	private Future<?> notificationWorker;
-	
-	private IdleStatusComponent idleComponent;
-	
-	private static int currentNotificationType;
-	private static CloudConnectionStatusEnum currentStatus = null;
-	
-	private static final HashSet<CloudConnectionStatusComponent> componentRegistry = new HashSet<CloudConnectionStatusComponent>();
-	
+	private GPIOService m_gpioService;
+
+	private int m_ledIndex = -1;
+	private KuraGPIOPin m_notificationLED;
+
+	private ExecutorService m_notificationExecutor;
+	private Future<?> m_notificationWorker;
+
+	private IdleStatusComponent m_idleComponent;
+
+	private static int m_currentNotificationType;
+	private static CloudConnectionStatusEnum m_currentStatus;
+
+	private static final HashSet<CloudConnectionStatusComponent> s_componentRegistry = new HashSet<CloudConnectionStatusComponent>();
```

Comment:
We should not use static fields (which incidentally should start with ``s_``) because if this component was not a singleton all instances will share the same state.

If I remove ``static`` and I disable/enable the ``GPIOService`` the component registry is lost and the led does not start blinking again (because components do not register again).

Wouldn't be better make the dependent components to have an optional dependency on the ``CloudConnectionStatusService`` and register when the CloudConnectionStatusService is injected by the SCR?
See also #445.